etl/scripts-ccao-data-warehouse-us-east-1/spatial/spatial-ccao-corner.R

```diff
@@ -0,0 +1,237 @@
+library(arrow)
+library(aws.s3)
+library(ccao)
+library(dplyr)
+library(geoarrow)
+library(glue)
+library(here)
+library(osmdata)
+library(purrr)
+library(sf)
+library(tictoc)
+
+# This script detects corner lots in Cook County parcels and saves a boolean
+# indicator as well as the cross used by the corner detection algorithm
+AWS_S3_RAW_BUCKET <- Sys.getenv("AWS_S3_RAW_BUCKET")
+AWS_S3_WAREHOUSE_BUCKET <- Sys.getenv("AWS_S3_WAREHOUSE_BUCKET")
+output_bucket <- file.path(AWS_S3_WAREHOUSE_BUCKET, "spatial", "ccao", "corner")
+
+# Get the parcel file years for which we should make corner lot indicators
+parcel_path <- file.path(AWS_S3_WAREHOUSE_BUCKET, "spatial", "parcel")
+parcel_years <- open_dataset(parcel_path) %>%
+  distinct(year) %>%
+  collect() %>%
+  pull(year)
+
+# Drop years before 2014, since OSM data for roads is spotty prior to that
+parcel_years <- parcel_years[parcel_years >= 2014]
+
+# Iterate over the years and townships, saving the results to a temporary file
+# after each iteration
+for (iter_year in parcel_years) {
```

Comment:
This new script runs by year and township, meaning we'll get corners for all years after and including 2014. It takes about 30 minutes per year to run.